### ISSUE TITLE
Consider: Allow filtering of packages in the Installed tab in NuGet PM UI based on package source drop down

### ISSUE THREAD

**Issue by mishra14** [NuGet/Home#5393]
Original Customer Request: https://github.com/NuGet/Home/issues/5340

I think we should consider this.

**Comment by mishra14** [comment]
Tagging PM folks for visibility: @anangaur @karann-msft

**Comment by emgarten** [comment]
Installed packages shows packages installed to your project. The source no longer matters at that point in my opinion, and trying to filter based on this causes the following problems:

* If a source is removed or a package is removed from a source the installed package would no longer appear unless an *unknown* option was added to the source list.
* Users do not expect to go to the installed tab and see an incomplete list, they would need to remember where packages came from in order to navigate to them.
* When offline the installed tab would be unable to show packages

Related: https://github.com/NuGet/Home/issues/1364

**Comment by emgarten** [comment]
Having an installed filter on the browse tab would make more sense to me than filtering the installed tab, but this extra knob would be confusing. Overall I think this is trying to solve a very specific and advanced scenario, it should be done in the package management console instead.

**Comment by jjanuszkiewicz** [comment]
(I'm the author of the #5340)

@emgarten The offline support and the case of a source being removed are very fair points. They give me a new perspective on this issue. In that case, what still remains is simply the fact that the behaviour of that source filter is confusing on Installed and Consolidate tabs. This is - from user's perspective - the same UI element as on Browse and Updates, but its function is different and it's not clear that its function changes as you navigate to a different tab.

I'm also not sure about the

> Users do not expect to go to the installed tab and see an incomplete list

point - I suppose you have data to support this, but the fact is that on my team, we've always considered it a minor but irritating bug (i.e. we expected the list to be incomplete depending on whether we selected All or a specific source).

I don't consider filtering installed packages to a single source a very specific or advanced scenario. I know that many companies have private package feeds. I imagine it is pretty common to want to see what company internal packages a given project uses (or the other way round, depending on the company specifics - what public packages from nuget.org a project uses).

Thinking about this some more - there is the All item on that dropdown, so you would still be able to see the full list. Maybe it could employ a hybrid approach - when All was selected, there would be full offline support as well as packages from unknown sources would be listed, while if a specific source was selected, packages would be limited to that source only? Wouldn't that be the best of both worlds?

**Comment by jjanuszkiewicz** [comment]
Whichever way this evolves, I think the UI could somehow indicate what the source filter does in each of the views. Even if its function doesn't get changed after all, it would prevent users (such as me) from thinking there is a bug :-) #5394

**Comment by emgarten** [comment]
> from user's perspective - the same UI element as on Browse and Updates, but its function is different and it's not clear that its function changes as you navigate to a different tab

I agree, I think this is the main problem with the UI. The installed tab is the same as browse, but using the project as the source, it's very confusing to have all the same options there.

**Comment by anangaur** [comment]
I agree to @emgarten that Package source filter means little on the "Installed" tab. The UI today may be confusing with the "Package source" visible with the "Installed" tab but it does some filtering on showing whether update is available or not for the feed selected as "Package source".

E.g.
If you have a PackageA installed with version 2.0.0 and in your local feed there is an updated version say 3.0.0 available but this is the latest version on nuget.org or any other source for your project, then the update option will only show up when you select "Package source" as "<Local feed>" or "All".
